src/geometry/manifold/smooth_manifold_with_corners.lean

```diff
@@ -703,45 +701,242 @@ not `local_homeomorph` as the target is not open in `E` in general, but we can s
 as `local_equiv`.
 -/
 
+namespace local_homeomorph
+/-- Given a chart `f` on a manifold with corners, `f.extend` is the extended chart to the model
+vector space. -/
+@[simp, mfld_simps] def extend : local_equiv M E :=
+f.to_local_equiv ≫ I.to_local_equiv
+
+lemma extend_coe : ⇑(f.extend I) = I ∘ f := rfl
+
+lemma extend_coe_symm : ⇑(f.extend I).symm = f.symm ∘ I.symm := rfl
+
+lemma extend_source : (f.extend I).source = f.source :=
+by rw [extend, local_equiv.trans_source, I.source_eq, preimage_univ, inter_univ]
+
+lemma is_open_extend_source : is_open (f.extend I).source :=
+by { rw extend_source, exact f.open_source }
+
+lemma extend_target : (f.extend I).target =
+  I.symm ⁻¹' f.target ∩ range I :=
```

Comment:
can you put the whole formula on one single line? (Either the first one or the second one depending on the 100 characters limit).